### ISSUE TITLE
update cwltool version and test monitoring options

### ISSUE THREAD

**Issue by denis-yuen** [ga4gh/dockstore#1915]
### Desired behaviour

As per https://gitter.im/ga4gh/dockstore?at=5bf7fdfab5154c550051df48 , for a hotfix, update and test a version of cwltool with monitoring tools against verified tools on Dockstore.org. 

Also seems to be missing html5lib? https://gitter.im/ga4gh/dockstore?at=5bf810cfa6e00567431e23de

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-435)
┆Issue Type: Story
┆Fix Versions: Dockstore 1.6
┆Sprint: Seabright Sprint 6 Ferry
┆Issue Number: DOCK-435

**Comment by denis-yuen** [comment]
Also need requests 2.20.0 or later

**Comment by denis-yuen** [comment]
Also, need to update pcawg workflows to be compatible with latest cwltool. 
See https://github.com/common-workflow-language/common-workflow-language/issues/674

**Comment by denis-yuen** [comment]
Would be nice, but not essential to be able to tear down testing framework and either bring it back up on collab or aws

**Comment by denis-yuen** [comment]
@garyluu will need to test with 20181217162649   https://gitter.im/common-workflow-language/common-workflow-language?at=5c658a7d604f233ab60c6566  (thanks @mr-c!)

As a bonus, I think the travis-ci tests for the versions around that date test with requests 2.20.0 or later.

**Comment by garyluu** [comment]
Update from tooltester after updating directly to cwltool version 20181217162649:
Problems with:


| Workflow | Immediate problem |
| --- | --- |
| #workflow/github.com/Barski-lab/ga4gh_challenge:master | Never finishes executing then out of memory |
| #workflow/github.com/DataBiosphere/topmed-workflows/u_of_Michigan_alignment_pipeline:1.31.0 | No longer exist and verified? |
| quay.io/pancancer/pcawg-sanger-cgp-workflow:2.0.3 | /var/spool/cwl missing |
| a bunch of bwa-mem workflows | /var/spool/cwl missing |
| quay.io/pancancer/pcawg_delly_workflow:2.0.1-cwl1.0 | /var/spool/cwl missing |

**Comment by NatalieEO** [comment]
accidentally assigned myself in jira

**Comment by denis-yuen** [comment]
Going back to the OP:
Looks like new cwltool version works
Looks like we can use the new Docker monitoring options
```
$ cat ~/.dockstore/config 
token: <snip>
server-url: https://staging.dockstore.org/api
#server-url: https://dockstore.org:443/api
cwltool-extra-parameters: --cidfile-dir, /home/ubuntu/foo, --cidfile-prefix, prefix
ubuntu@testing-client:~$ dockstore tool launch --entry quay.io/briandoconnor/dockstore-tool-md5sum:1.0.4 --json Dockstore.json
Downloading: file:///home/ubuntu/Dockstore.json to file:///tmp/parameter2368939667934806066json
[##################################################] 100%
Created temporary working directory at '/tmp/1553716879258-0'
Successfully downloaded files for entry 'quay.io/briandoconnor/dockstore-tool-md5sum'
Creating directories for run of Dockstore launcher at: /home/ubuntu/datastore//launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f
Provisioning your input files to your local machine
Preparing download location for: #input_file from md5sum.input into directory: /home/ubuntu/datastore//launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/inputs/3e68b7ff-7e9f-413d-afb0-617c5d6515bc
WARNING: Output location not found for "output_file" provisioning by default to working directory
Calling out to cwltool to run your workflow
########### WARNING ###########
You are using extra flags for your cwl runner which may not be supported. Use at your own risk.
--cidfile-dir
/home/ubuntu/foo
--cidfile-prefix
prefix
Executing: cwltool --cidfile-dir /home/ubuntu/foo --cidfile-prefix prefix --enable-dev --non-strict --outdir /home/ubuntu/datastore//launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/outputs/ --tmpdir-prefix /home/ubuntu/datastore//launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/tmp/ --tmp-outdir-prefix /home/ubuntu/datastore//launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/working/ /tmp/1553716879258-0/Dockstore.cwl /home/ubuntu/datastore/launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/workflow_params.json
/home/ubuntu/.local/bin/cwltool 1.0.20181217162649
Resolved '/tmp/1553716879258-0/Dockstore.cwl' to 'file:///tmp/1553716879258-0/Dockstore.cwl'
[job Dockstore.cwl] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.
<snip>
Final process status is success

Saving copy of cwltool stdout to: /home/ubuntu/datastore/launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/outputs/cwltool.stdout.txt
Saving copy of cwltool stderr to: /home/ubuntu/datastore/launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/outputs/cwltool.stderr.txt
Provisioning your output files to their final destinations
Registering: #output_file to provision from /home/ubuntu/datastore/launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/outputs/md5sum.txt to : .
Provisioning from /home/ubuntu/datastore/launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/outputs/md5sum.txt to ./md5sum.txt
Downloading: file:///home/ubuntu/datastore/launcher-df58e0df-5e2e-48e5-81e9-dcf6cc62089f/outputs/md5sum.txt to file:///home/ubuntu/md5sum.txt
[##################################################] 100%
ubuntu@testing-client:~$ ls foo
prefix-20190327195940-389299.cid  prefix-20190327200123-923767.cid
```
